src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -90,16 +57,57 @@ public class BatchHandler : DatabaseRequestHandler
 
                 HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
 
-                using (var writer = new BlittableJsonTextWriter(ctx, ResponseBodyStream()))
+                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
                 {
-                    ctx.Write(writer, new DynamicJsonValue
+                    context.Write(writer, new DynamicJsonValue
                     {
                         ["Results"] = mergedCmd.Reply
                     });
                 }
             }
         }
 
+        private async Task<MergedBatchCommand> CreateBatchCommand(DocumentsOperationContext context)
+        {
+            var command = new MergedBatchCommand {Database = Database};
+
+            if (HttpContext.Request.ContentType != null &&
+                HttpContext.Request.ContentType.StartsWith("multipart/mixed", StringComparison.OrdinalIgnoreCase))
```

Comment:
And this is till a rare thing, so we want the common thing (just parsing the request to be the first thing here, then the request with attachment to be in a separate method.